torch/_functorch/aot_autograd.py

```diff
@@ -2851,9 +2846,54 @@ def functional_call(*args, **kwargs):
         bw_compiler = fw_compiler
 
     full_args = []
+    # First, the params
     full_args.extend(params_flat)
+
+    aot_autograd_arg_pos_to_source = []
+    # Then, the params 1:1 mapped sources, if relevant.
+    if hasattr(mod, "_nn_module_sources"):
+        # We now know this came from dynamo, and (1) we care about guards,
+        # so setting up aot_autograd_arg_pos_to_source for downstream dedup guards
+        # can now be done safely. (2) Dynamo logic protects the 1:1 sizing below.
+        for name in params.keys():
+            if name in mod._nn_module_sources:
+                # Easy case
+                aot_autograd_arg_pos_to_source.append(mod._nn_module_sources[name])
+            else:
+                # Complex case - an attribute is found with "." in its name!
+                # This is not valid for dynamo named params, so, we need to reconstruct
+                # the source.
+                # A name is always a base_name, and a param_name
+                base_name, param_name = name.split(".")
+                # The base name should always be known
+                base = mod._nn_module_sources[base_name]
+                # TODO(voz): Move this to torch._guards
+                from torch._dynamo.source import AttrSource
+                # Reconstruct this to ensure we get the same name contract as dynamo
+                new_source = AttrSource(base, param_name)
+                # Sanitize the name, just as dynamo does
+                new_name = new_source.name()
+                # BEGIN DUPLICATIVE LOGIC TO DEDUP
+                new_name = re.sub(r"\[(\d+)\]", r"_\g<1>", new_name)
+                new_name = re.sub(r"[^a-zA-Z0-9]", "_", new_name)
+                if not new_name[0].isalpha():
+                    new_name = "sub" + new_name
+                # END DUPLICATIVE LOGIC TO DEDUP
```

Comment:
I am leaving it in this PR as such, for now, to make it easier to understand.  The 3 lines of duplicate code could ostensibly be moved to a util, but it is not overly important.

Reply:
why not do the util now actually?  seems equally hard to write the explanation and todo haha